add some effects for play screen

fcebk17 opened the pull request:
## Types of changes
**Thanks for sending a pull request! Please fill in the following content to let us know better about this change.**
Please put an `x` in the box that applies

- [ ] **Bugfix**
- [x] **New feature**
- [ ] **Refactoring**
- [ ] **Breaking change** (any change that would cause existing functionality to not work as expected)
- [ ] **Documentation Update**
- [ ] **Other (please describe)**

## Description
**In this version, I only create some effect on website, websocket not connected**

+ Click heart image will trigger lightning and thunder music
+ Click heart image will turn into broken heart </3
+ Add a rocket pic to the index page

## Steps to Test This Pull Request
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

## Expected behavior
A clear and concise description of what you expected to happen.

## Related Issue
If applicable, refernce to the issue related to this pull request.

## More Information
# Index 用27吋螢幕縮放100%
![image](https://user-images.githubusercontent.com/83528127/182975613-c1bc320b-52b1-403e-b6cb-091bbbd77071.png)

fcebk17 committed:
add lightning effect

in this version, I only create some effect on website, websocket not connected

Change:
+ Click heart image  will trigger lightning and thunder music
+ Click heart image will turn into broken heart </3
+ Add a rocket pic to the index page

asas1asas200 reviewed (changes requested):
Resolve the conflicts before you request a review.
Read [this](https://juejin.cn/post/6844903648208617485) and rebase the trivial commits.

fcebk17 commented:
Conflict resolved

asas1asas200 commented on templates/base.html:
```diff
@@ -7,8 +7,11 @@
   <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='/css/background.css') }}" >
   <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='/css/button.css') }}" >
   <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='/css/popup_frame.css') }}" >
+  <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='/css/lightning.css') }}" >
+  <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='css/animate.min.css')}}">
```
If these files are optional needed, include them in `{% block head %}` in specific page.

fcebk17 replied:
done

asas1asas200 commented on templates/base.html:
```diff
@@ -7,8 +7,11 @@
+  <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='css/animate.min.css')}}">
   <script src="{{ url_for('static', filename='/js/bootstrap.bundle.min.js') }}"></script>
   <script src="{{ url_for('static', filename='/js/playbgm.js') }}"></script>
+  <script src="{{ url_for('static', filename='/js/HitPoint.js') }}"></script>
```
This file also

fcebk17 replied:
done

asas1asas200 commented on templates/play.html:
```diff
@@ -26,11 +26,48 @@
         <div class="row">
+          <div class = "col"></div>
+          <div class = "col-1" >
+            <img id="heart" src="{{url_for('static', filename='/images/heart.png')}}" widht="80" height="80" id="heart" onclick="changeImage()">
+          </div>
+          <div class = "col-1">
+            <img id="heart1" src="{{url_for('static', filename='/images/heart.png')}}" widht="80" height="80" id="heart1" onclick="changeImage1()">
+          </div>
+          <div class = "col-1">
+            <img id="heart2" src="{{url_for('static', filename='/images/heart.png')}}" widht="80" height="80" id="heart2" onclick="changeImage2()">
+          </div>
+          <div class = "col-1">
+            <img id="heart3" src="{{url_for('static', filename='/images/heart.png')}}" widht="80" height="80" id="heart2" onclick="changeImage3()">
+          </div>
+          <div class = "col-1">
+            <img id="heart4" src="{{url_for('static', filename='/images/heart.png')}}" widht="80" height="80" id="heart2" onclick="changeImage4()">
+          </div>
```
Too many heart id `heart2`.

Rewrite this by jinja2 templates for-loop to avoid the same code.

fcebk17 replied:
Ok, I will change it

fcebk17 replied:
done

asas1asas200 replied:
Commit and push it to the repo.

asas1asas200 commented on static/js/HitPoint.js:
```diff
@@ -0,0 +1,78 @@
+    if (element.src.match('heart')){
+        element.src ="/static/images/broken-heart.png";
+        const anime = document.querySelector('.crack2');
+        anime.classList.add('animate__animated', 'animate__flash');
+        const photo = document.querySelector('.photo');
+        photo.classList.add('animate__animated', 'animate__shakeX')
+        const music = new Audio('static/bgm/Thunder-crack.mp3');
+        music.play();
+    }
+    else{
+        element.src="/static/images/heart.png";
+    }
+
+}
+
+function changeImage3(){
+    element = document.getElementById('heart3');
+    const anime = document.querySelector('.crack3');
+    if (element.src.match('heart')){
+        element.src ="/static/images/broken-heart.png";
+        anime.classList.add('animate__animated','animate__flash');
+        const music = new Audio('static/bgm/Thunder-crack.mp3');
+        music.play();
+    }
+    else{
+        element.src="/static/images/heart.png";
+    }
+
+}
+
+function changeImage4(){
+    element = document.getElementById('heart4');
+    const anime = document.querySelector('.crack4');
+    if (element.src.match('heart')){
+        element.src ="/static/images/broken-heart.png";
+        anime.classList.add('animate__animated', 'animate__flash');
+        const music = new Audio('static/bgm/Thunder-crack.mp3');
+        music.play();
+    }
+}
```
These functions are almost the same, why don't pass a parameter into the funciton.

fcebk17 replied:
# FIXME

asas1asas200 committed:
refactor(play): rewrite animations on play page

Rename demo.js to play.js.
Rewrite HTML tags of heart and lightning.
    The {{ for n in range(5) }} can be edited to add more hearts in the
    game.
    The flashes of lightning now appear randomly.
Remove shaking animation on webcam, it can be added back if needed.
Remove animate.min.css due to unuse.

asas1asas200 reviewed (approved):
@fcebk17 
I rewrited something, check if it works.
Merge it if no problems.
